source/grid/manifold_lib.cc

```diff
@@ -1009,16 +1011,19 @@ ::get_possible_cells_around_points(const std::vector<Point<spacedim> > &points)
       if (&cell->get_manifold() != this)
         continue;
 
+      const std::array<Point<spacedim>, GeometryInfo<2>::vertices_per_cell> vertices
+      {{cell->vertex(0), cell->vertex(1), cell->vertex(2), cell->vertex(3)}};
+
       // cheap check: if any of the points is not inside a circle around the
       // center of the loop, we can skip the expensive part below (this assumes
       // that the manifold does not deform the grid too much)
       Point<spacedim> center;
       for (unsigned int v=0; v<GeometryInfo<dim>::vertices_per_cell; ++v)
-        center += cell->vertex(v);
+        center += vertices[v];
```

Comment:
What happens with that code in 3D? Aren't we here at a generic dimension and would need to put 8 points in 3D?